### DISCUSSION TITLE
When the taskbar groupping is enabled, the taskbar buttons are too big in Classic theme

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#478] [General]
![изображение](https://user-images.githubusercontent.com/1271736/143622160-e66e3ea0-d7ee-4e23-a839-21e872002707.png)

Possibly, something should limit the size of the buttons in this mode.

Also, there is transparency and alignment issue in the flyer:
![изображение](https://user-images.githubusercontent.com/1271736/143622023-3acadee5-d82c-46e3-964a-5e9cff8bceb4.png)

Also, the move list (right mouse button on taskbar item) is immersive, Windows 11-themed. Is there a way to make it non-immesive, like context menus? Cannot make a screenshot of it.

**Comment by valinet** [comment]
> Also, the move list (right mouse button on taskbar item) is immersive, Windows 11-themed. Is there a way to make it non-immesive, like context menus? Cannot make a screenshot of it.

Use 7+TT to have it display the system menu instead.

> Possibly, something should limit the size of the buttons in this mode.
> Also, there is transparency and alignment issue in the flyer:

Yeah, has been like this since it was released, I mean, someone could look into it at some point... who uses classic theme without labels anyway...? I mean, whoever uses it, has more incentive to maybe fix it, I guess...

**Reply by Anixx** [reply]
> Use 7+TT to have it display the system menu instead.

I do so. But somebody may like it this way. I can tell that on Windows 8.1 this thing was themed Classic under Classic theme.

**Reply by Anixx** [reply]
> Yeah, has been like this since it was released,

Initially, the buttons were of unlimited size even in non-groupped mode, and you fixed it. So, I thought, fixing it for groupped mode can be similar.

**Reply by Anixx** [reply]
Also, I wonder, could the window previews (thumbnails) on taskbar were visible in Classic theme as well? By now they are automatically replaced by the group lists when Classic theme is active.

**Reply by valinet** [reply]
Idk, study the disassembly, all I know is from things in the disassembly that I studied, other than that, I haven't and do not currently work for Microsoft, I never talked to them directly, never saw their source code, so I can't tell anything about things I haven't touched on as I have no other source of information. In general, anything is possible, that's one of my mottos. It's also a costly motto, if I may add.

> Initially, the buttons were of unlimited size even in non-groupped mode, and you fixed it. So, I thought, fixing it for groupped mode can be similar.

Idk, I played with some `uxtheme` functions that seemed to offer the metrics for those buttons, specifically `GetThemeMargins` if I remember well. Maybe when grouped, something else's used, idk. Test and tell me. Honestly, I don't get why you don't get down to looking into this yourself, especially considering how of interest this area is for you... or this classic theme community in general, the only way to properly fix this is to get down to things like this, inevitably, some things really are broken, no one bothered to implement them with lack of theming support in mind, some years have passed since 2009 and they haven't been entirely sleeping in all these years.

**Comment by valinet** [comment]
Anyway, it's not the first time I am telling you, make a SINGLE thread, IDEALLY use the existing discussion about the classic theme related things that we had initially (https://github.com/valinet/ExplorerPatcher/discussions/101), and post all things like these there. It's easier like that, it's easier to go into one place and find out what one can work on, what's the current status etc rather than have everything scattered throughout a bunch of posts where it's mostly me that I am replying anyway.

Thanks
